src/hotspot/share/opto/loopnode.cpp

```diff
@@ -3571,21 +3576,31 @@ void PhaseIdealLoop::collect_potentially_useful_predicates(
 // Note: it will also eliminates loop limits check predicate since it also uses
 // Opaque1 node (see Parse::add_predicate()).
 void PhaseIdealLoop::eliminate_useless_predicates() {
-  if (C->predicate_count() == 0)
+  if (C->predicate_count() == 0 && C->skeleton_predicate_count() == 0) {
     return; // no predicate left
+  }
 
   Unique_Node_List useful_predicates; // to store useful predicates
   if (C->has_loops()) {
     collect_potentially_useful_predicates(_ltree_root->_child, useful_predicates);
   }
 
   for (int i = C->predicate_count(); i > 0; i--) {
-     Node * n = C->predicate_opaque1_node(i-1);
+     Node* n = C->predicate_opaque1_node(i - 1);
      assert(n->Opcode() == Op_Opaque1, "must be");
      if (!useful_predicates.member(n)) { // not in the useful list
        _igvn.replace_node(n, n->in(1));
      }
   }
+
+  for (int i = C->skeleton_predicate_count(); i > 0; i--) {
+    const int idx = i - 1;
```

Comment:
Could be removed for consistency with above code.

Reply:
Yes, that was a left over from a previous commit where we needed `idx` twice. I update that.